internal/sink/sinkSender.go

```diff
@@ -94,7 +96,7 @@ type SinkMetrics struct {
 func NewSinkSender(sw *SinkWrapper, l Listener) (sender *SinkSender, err error) {
```

Comment:
```suggestion
func NewSender(sw *SinkWrapper, l Listener) (sender *Sender, err error) {
```

general note, we should consider renaming things with `sink` as their prefix since now we have the `sink` package

a note about interfaces and structs that may share the same name, assuming there's a need for the given interface, I recommend exposing the interface and privating the struct with the same name


Example:

```go
type Sender interface {
	Update(Listener) error
	Shutdown(bool)
	RetiredSince() time.Time
	Queue(*wrp.Message)
}

type sender struct {
}
```